src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -131,8 +134,46 @@ public CustomObjectServiceImpl(@Nonnull final CustomObjectSyncOptions syncOption
         if (StringUtils.isEmpty(customObjectDraft.getKey()) || StringUtils.isEmpty(customObjectDraft.getContainer())) {
             return CompletableFuture.completedFuture(Optional.empty());
         }
-        return createResource(customObjectDraft,
-            draft -> CustomObjectCompositeIdentifier.of(draft).toString(),
-            CustomObjectUpsertCommand::of);
+        CompletionStage<Optional<CustomObject<JsonNode>>> createdResource = createResource(customObjectDraft,
+            draft -> CustomObjectCompositeIdentifier.of(draft).toString(), CustomObjectUpsertCommand::of);
+        return createdResource ;
+    }
+
+    /**
+     * This method overrides {@link BaseService#executeCreateCommand}. It is because custom object has different
+     * behaviour from other resource in error handling, which requires to be handled separately.
```

Comment:
We can remove these 2 lines. I think this is obvio

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/573/commits/73bda63444316f879ab7c436391c31f8fc3ab069